redis/src/sentinel.rs

```diff
@@ -0,0 +1,558 @@
+    replica_start_index: usize,
+}
+
+fn sentinel_masters_cmd() -> crate::Cmd {
+    let mut cmd = crate::cmd("SENTINEL");
+    cmd.arg("MASTERS");
+    cmd
+}
+
+fn sentinel_replicas_cmd(master_name: &str) -> crate::Cmd {
+    let mut cmd = crate::cmd("SENTINEL");
+    cmd.arg("SLAVES"); // For compatibility with older redis versions
+    cmd.arg(master_name);
+    cmd
+}
+
+#[cfg(feature = "aio")]
+async fn async_try_sentinel_masters(
+    connection_info: &ConnectionInfo,
+) -> RedisResult<Vec<HashMap<String, String>>> {
+    let sentinel_client = Client::open(connection_info.clone())?;
+    let mut conn = sentinel_client.get_async_connection().await?;
+    let result: Vec<HashMap<String, String>> =
+        sentinel_masters_cmd().query_async(&mut conn).await?;
+    Ok(result)
+}
+
+#[cfg(feature = "aio")]
+async fn async_try_sentinel_replicas(
+    connection_info: &ConnectionInfo,
+    master_name: &str,
+) -> RedisResult<Vec<HashMap<String, String>>> {
+    let sentinel_client = Client::open(connection_info.clone())?;
+    let mut conn = sentinel_client.get_async_connection().await?;
+    let result: Vec<HashMap<String, String>> = sentinel_replicas_cmd(master_name)
+        .query_async(&mut conn)
+        .await?;
+    Ok(result)
+}
+
+fn try_sentinel_masters(
+    connection_info: &ConnectionInfo,
+) -> RedisResult<Vec<HashMap<String, String>>> {
+    let sentinel_client = Client::open(connection_info.clone())?;
+    let mut conn = sentinel_client.get_connection()?;
+    let result: Vec<HashMap<String, String>> = sentinel_masters_cmd().query(&mut conn)?;
+    Ok(result)
+}
+
+fn try_sentinel_replicas(
+    connection_info: &ConnectionInfo,
+    master_name: &str,
+) -> RedisResult<Vec<HashMap<String, String>>> {
+    let sentinel_client = Client::open(connection_info.clone())?;
+    let mut conn = sentinel_client.get_connection()?;
+    let result: Vec<HashMap<String, String>> =
+        sentinel_replicas_cmd(master_name).query(&mut conn)?;
+    Ok(result)
+}
+
+fn is_master_valid(master_info: &HashMap<String, String>, service_name: &str) -> bool {
+    master_info.get("name").map(|s| s.as_str()) == Some(service_name)
+        && master_info.contains_key("ip")
+        && master_info.contains_key("port")
+        && master_info.get("flags").map_or(false, |flags| {
+            flags.contains("master") && !flags.contains("s_down") && !flags.contains("o_down")
+        })
+        && master_info["port"].parse::<u16>().is_ok()
+}
+
+fn is_replica_valid(replica_info: &HashMap<String, String>) -> bool {
+    replica_info.contains_key("ip")
+        && replica_info.contains_key("port")
+        && replica_info.get("flags").map_or(false, |flags| {
+            !flags.contains("s_down") && !flags.contains("o_down")
+        })
+        && replica_info["port"].parse::<u16>().is_ok()
+}
+
+fn random_replica_index(max: usize) -> usize {
+    if max == 0 {
+        0
+    } else {
+        let nanos = SystemTime::now()
```

Comment:
Sorry, I should have commented (both in the PR and the code) why I did this. I mentioned this in my reply to you last comment below, but basically I wanted to avoid an extra dependency just for this, since it seems rather simple and harmless enough that any seemingly random number would suffice. Specially if this isn't going to be behind a feature flag. Does that make sense? Or am I being overzealous about adding new dependencies?

Reply:
both `oorandom`, `rand` and `fastrand` are used by in redis-rs dependency tree.
IMO, there's no problem with adding a direct dependency on one of them, but I'll give @jaymell the final word on this.

Reply:
Yeah, we already use `rand` elsewhere, we can just make it non-optional. 

For what it's worth, I think it's also considering making the sentinel functionality itself a feature. It would be easy to do, since the code is so self-contained.

Reply:
@jaymell any workaround to connect just to master sentinel node using your library? If I just specify protocol as `redis-sentinel` in a connection string, the thing panics. Would be glad for some solution ad-hoc until this is merged.